src/commands.rs

```diff
@@ -5138,6 +5143,36 @@ fn cmd_git_push(
         }
         tx = workspace_command
             .start_transaction(&format!("push all branches to git remote {}", &args.remote));
+    } else {
+        match workspace_command
+            .repo()
+            .view()
+            .get_checkout(&workspace_command.workspace_id())
+        {
+            None => {
+                return Err(UserError("No current branch.".to_string()));
+            }
+            Some(checkout) => {
+                let desired_target = Some(RefTarget::Normal(checkout.clone()));
+                for (branch_name, branch_target) in workspace_command.repo().view().branches() {
+                    if branch_target.local_target == desired_target {
+                        let push_action = classify_branch_push_action(branch_target, &args.remote);
+                        match push_action {
+                            BranchPushAction::AlreadyMatches => {}
+                            BranchPushAction::LocalConflicted => {}
+                            BranchPushAction::RemoteConflicted => {}
+                            BranchPushAction::Update(update) => {
+                                branch_updates.push((branch_name.clone(), update));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        tx = workspace_command.start_transaction(&format!(
```

Comment:
It's probably better to warn if `branch_updates` is empty here. "Nothing changed" doesn't make sense.

I occasionally do `jj branch set -r @- master && jj git push` to push the current-ish branch.

Reply:
Good point! Fixed.